primitives/state-machine/src/lib.rs

```diff
@@ -945,23 +934,24 @@ mod execution {
 		let proving_backend =
 			TrieBackendBuilder::wrap(trie_backend).with_recorder(recorder.clone()).build();
 		let mut count = 0;
-		proving_backend
-			.apply_to_key_values_while(
-				child_info,
+		let iter = proving_backend
+			.keys(IterArgs {
```

Comment:
```suggestion
			.pairs(IterArgs {
```

Reply:
These values aren't actually used in the loop though, so isn't this just a waste to also fetch the values?

Reply:
It should change the size of the proof.

Reply:
@cheme I might be misunderstanding something here, but I've added extra asserts which check the generated proof sizes (with values generated *before* this PR and my previous PR) and after applying this PR the test still passes, so it looks like the size of the proof remains unchanged? What am I missing? Am I doing something wrong here? Any hints on how I can test this?

Reply:
The trie you iterate on need to be V1 and values in tree needs to be of size > 32 bytes (so key iterator only include their hashes in the proof).

Reply:
actually the method really need to be iterating on key and values, it is important that values are included in the proof (used to sync state in after a warp sync).

Reply:
Thanks! I'll try to do that.

> actually the method really need to be iterating on key and values

Yeah, I'm not really doubting the necessity of doing this. I just really don't want to make this change blindly without leaving behind a test behind as the next person changing this code could make the same mistake and this is really subtle trap and currently (as evidenced by the green CI) not tested at all.

Reply:
Okay, I've fixed the issue, added a comment, and added a test.

I've ran the test before all my changes (both PRs) and it passes. After this PR without the fix it fails. With the fix it passes. So now this is actually tested.

@cheme Thanks for your help!

Reply:
you're very welcome (this is hard to guess and a bit undertested).